src/Core/Managed/Shared/Channel/InMemoryChannel.cs

```diff
@@ -133,6 +133,16 @@ public void Send(ITelemetry item)
                 return;
             }
 
+            if (string.IsNullOrEmpty(item.Context.InstrumentationKey))
+            {
+                if (CoreEventSource.Log.IsVerboseEnabled)
+                {
+                    CoreEventSource.Log.ItemRejectedNoInstrumentationKey(item.ToString());
```

Comment:
item.ToString() will output item type (or at least it does for TraceTelemetry). Do you think this is good enough?